File: aug22_20/at4.py


if __name__ == "__main__":
    n = int(input())
    x =list((int(x) for x in input().split()))
    # print(x)
    
    tmp1 = 0
    tmp2 = x[0]
    for i in range(1,n):
        if tmp2 > x[i]:
            tmp3 = tmp2 - x[i]
            tmp1 += tmp3

            x[i] = x[i] + tmp3
            tmp2 = x[i]
        else:
            tmp1 += 0
            tmp2 = x[i]

    print(tmp1)

File: aug22_20/at4.cpp
#include <iostream>
#include <vector>
#include <iterator>

using namespace std;


int main(){
    int n;
    cin >> n;
    int input;
    cin >> input;
    vector<int> tmp1;
    tmp1.push_back(input);

    int tmp2 = tmp1[0];
    int tmp3 = 0;
    for(int i=1; i<n; i++){
        if(tmp1[i]>tmp2){
            tmp3 += tmp1[i] - tmp2;
            tmp2 = tmp1[i] + tmp3;
        }
    }

    cout << tmp3 << endl;

}